src/main/java/com/limechain/runtime/hostapi/StorageHostFunctions.java

```diff
@@ -285,12 +287,23 @@ public RuntimePointerSize extStorageChangesRootVersion1(RuntimePointerSize paren
      * @return a pointer-size to the SCALE encoded Option value containing the next key in lexicographic order.
      */
     public RuntimePointerSize extStorageNextKeyVersion1(RuntimePointerSize keyPointer) {
-        byte[] key = hostApi.getDataFromMemory(keyPointer);
-        byte[] nextKey = repository.getNextKey(new String(key)).map(String::getBytes).orElse(null);
+        Nibbles key = Nibbles.fromBytes(hostApi.getDataFromMemory(keyPointer));
+        byte[] nextKey = repository.getNextKey(key)
+                .map(NibblesUtils::toBytesAppending)
+                .map(this::asByteArray)
+                .orElse(null);
 
         return hostApi.writeDataToMemory(scaleEncodedOption(nextKey));
     }
 
+    private byte[] asByteArray(List<Byte> bytes) {
```

Comment:
I could suggest [this approach](https://github.com/LimeChain/Fruzhin/blob/353de481b99e28490255b5c6d9a424f6203369ed/src/main/java/com/limechain/trie/dto/node/DecodedNode.java#L217) as an alternative to this helper method. Only as a possibility if you prefer it.

Reply:
The suggested approach seems to traverse the array twice, doesn't it?

Reply:
Yes. Depends on whether we value code reduction over performance or not. Was just a suggestion, rather nitpicky. 😄